helpers/base.sh

```diff
@@ -196,28 +196,40 @@ check_python_version() {
     fi
 }
 
-# Chooses a suitable compressor depending on which parameters passed (or didn't
+# Execute a suitable compression depending on which parameters passed (or didn't
 # pass) the user.
 # Globals:
 #     COMPRESS_WITH_BZIP2
 #     COMPRESS_WITH_GZIP
 #     COMPRESS_WITH_XZ
+#     IMAGE
+#     IMAGE_OWNERSHIP
 # Arguments:
 #     None
 # Returns:
-#     String which consists of the executable name and the extension name
-choose_compressor() {
+#     None
+compress_image() {
+    local extension=""
+
     if ${COMPRESS_WITH_BZIP2}; then
-        echo "bzip2 .bz2"
+        info "compressing image using bzip2"
+        compress_with_bzip2
+        extension=".bz2"
     fi
 
     if ${COMPRESS_WITH_GZIP}; then
-        echo "gzip .gz"
+        info "compressing image using gzip"
+        gzip "${IMAGE}"
+        extension=".gz"
     fi
 
     if ${COMPRESS_WITH_XZ}; then
-        echo "xz .xz"
+        info "compressing image using xz"
+        compress_with_xz
+        extension=".xz"
     fi
+
+    chown "${IMAGE_OWNERSHIP}" "${IMAGE}${extension}"
```

Comment:
I guess it's a bad idea to transform `compress_image` into a function with a side-effect.
Make the function return the path to the file with its new extension. Then, invoke `chown` outside of the function.